### PR TITLE
Upgrade to JHipster v6

### PR DESCRIPTION
**Pull request by vivekmore**
- Upgrade to JHipster v6
- Fix generated index.js that currently doesn't pass linting

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
Thanks @vivekmore !
I'll try to have a look next week

**Comment by pascalgrimaud** [comment]
Busy the last days, I'll review asap, I don't forget this PR

File: test/test-app.js [hunk 1]
```diff
@@ -49,6 +50,16 @@ describe('JHipster generator module', () => {
         it('generates default files', () => {
             assert.file(expectedFiles.module);
         });
+        it('generates index.js with correct content', () => {
+            const actualContent = fs.readFileSync('generators/app/index.js', 'utf8');
+            const expectedContent = fs.readFileSync(`${__dirname}/results/index.js.txt`, 'utf8');
```
Comment (murdos) [inline test/test-app.js]: I'm a bit worried that these result files will need to be (manually?) updated for each minor change done in the generator.
Still, this is an interesting approach, that allows you to visualize the expected content.

@pascalgrimaud : what do you think?

Comment (pascalgrimaud) [reply test/test-app.js]: Same comment like you @murdos 
The files inside `results/` folder need to be updated for each changes, so I don't think it's a good idea to test the content

Comment (vivekmore) [reply test/test-app.js]: In my opinion that is a good way to ensure that your expected generated content changes with your production code changes.

Currently there's no way to check that without actually running the generator against a prebuilt app.

I believe this automated test relieves one from having to do so every time there's a change.

Comment (murdos) [reply test/test-app.js]: @vivekmore could you remove the result files and related tests from this PR? That will allow to merge this PR and the fixes it contains.

Comment (murdos) [reply test/test-app.js]: Do you think you'll be able to update this PR soon? We need that to release a new version compatible with JHipster 6...

Comment (vivekmore) [reply test/test-app.js]: Sorry busy at the moment.

**Comment by murdos** [comment]
I've removed result files and related checks, so I'm merging this PR.
Thanks @vivekmore :)

**Pull request merged**